shell/browser/electron_browser_context.cc

```diff
@@ -163,6 +166,15 @@ ElectronBrowserContext::~ElectronBrowserContext() {
                             std::move(resource_context_));
 }
 
+content::StoragePartition*
+ElectronBrowserContext::GetDefaultStoragePartition() {
+  if (quota_size) {
+    return GetStoragePartition(
+        content::StoragePartitionConfig::CreateDefault(this, quota_size));
+  }
+  return BrowserContext::GetDefaultStoragePartition();
+}
```

Comment:
How about this:
```
content::StoragePartition* partition = BrowserContext::GetDefaultStoragePartition();
partition->GetQuotaManager()->SetQuotaSettings(storage::GetHardCodedSettings(quota));
return partition;
```
You can then avoid modifying StoragePartitionConfig.

Reply:
I had a try with these changes.  The SetQuotaSettings() member exists within the QuotaManagerImpl which is not exposed to via  GetQuotaManager(). Additionally,

Reply:
> I had a try with these changes. To have SetQuotaSettings() callable, the header file storage/browser/quota/quota_manager.h had to be added.
> 
> Since the SetQuotaSettings call can be done only on the IOThread, I tried the snippet below, but then there was a DCHECK fault since the IOthread is not accessible within the electron layer.
> 
> ```
>  content::StoragePartition* partition = BrowserContext::GetDefaultStoragePartition();
>  base::WeakPtrFactory<storage::QuotaManager> quota_weak_factory_{partition->GetQuotaManager()};
>  content::GetIOThreadTaskRunner({})->PostTask(
>           FROM_HERE, base::BindOnce(&storage::QuotaManager::SetQuotaSettings,
>           quota_weak_factory_.GetWeakPtr(),storage::GetHardCodedSettings(quota_size)));
> ```

I had a try with these changes.  To have SetQuotaSettings()  callable, the header file storage/browser/quota/quota_manager.h had to be added.

Since the SetQuotaSettings call can be done only on the IOThread, I tried the snippet below, but then there was a DCHECK fault since the IOthread is not accessible within the electron layer.

```
 content::StoragePartition* partition = BrowserContext::GetDefaultStoragePartition();
 base::WeakPtrFactory<storage::QuotaManager> quota_weak_factory_{partition->GetQuotaManager()};
 content::GetIOThreadTaskRunner({})->PostTask(
          FROM_HERE, base::BindOnce(&storage::QuotaManager::SetQuotaSettings,
          quota_weak_factory_.GetWeakPtr(),storage::GetHardCodedSettings(quota_size)));
```

Reply:
> How about this:
> 
> ```
> content::StoragePartition* partition = BrowserContext::GetDefaultStoragePartition();
> partition->GetQuotaManager()->SetQuotaSettings(storage::GetHardCodedSettings(quota));
> return partition;
> ```
> 
> You can then avoid modifying StoragePartitionConfig.

I had a try with these changes. To have SetQuotaSettings() callable, the header file storage/browser/quota/quota_manager.h had to be added.

Since the SetQuotaSettings call can be done only on the IOThread, I tried the snippet below, but then there was a DCHECK fault since the IOthread is not accessible within the electron layer.

 content::StoragePartition* partition = BrowserContext::GetDefaultStoragePartition();
 base::WeakPtrFactory<storage::QuotaManager> quota_weak_factory_{partition->GetQuotaManager()};
 content::GetIOThreadTaskRunner({})->PostTask(
          FROM_HERE, base::BindOnce(&storage::QuotaManager::SetQuotaSettings,
          quota_weak_factory_.GetWeakPtr(),storage::GetHardCodedSettings(quota_size)));

Reply:
What is the exact DCHECK error? ElectronBrowserContext::~ElectronBrowserContext() has these lines:
`BrowserThread::DeleteSoon(BrowserThread::IO, FROM_HERE,
std::move(resource_context_));`